src/CoreWCF.MSMQ/src/CoreWCF/Channels/MsmqBindingElementBase.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using CoreWCF.Queue.Common.Configuration;
+using CoreWCF.Runtime;
+
+namespace CoreWCF.Channels
+{
+    public abstract class MsmqBindingElementBase : QueueBaseTransportBindingElement
+    {
+        private DeadLetterQueue _deadLetterQueue;
+        private int _maxRetryCycles;
+        private int _receiveRetryCount;
+        private TimeSpan _retryCycleDelay;
+        private TimeSpan _timeToLive;
+
+        //private ReceiveErrorHandling receiveErrorHandling;
+
+        public Uri CustomDeadLetterQueue { get; set; }
+
+        public DeadLetterQueue DeadLetterQueue
+        {
+            get { return _deadLetterQueue; }
+            set
+            {
+                if (!DeadLetterQueueHelper.IsDefined(value))
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException("value"));
+                }
```

Comment:
Can you replace string quoted parameter names with nameof. So here it would be `nameof(value)`. There's another one on line 52. These don't get discovered easily by the csharp source analyzers. Here is a useful list of find/replace pairs that I use in VS for doing these more easily. You need to turn on the use regex option on the find, but this makes life a lot easier.
```
ThrowHelperError\(new ArgumentNullException\(\"([a-zA-Z0-9]+)\"\)
ThrowHelperError(new ArgumentNullException(nameof($1))

ThrowHelperError\(new ArgumentOutOfRangeException\(\"([a-zA-Z0-9]+)\"\)
ThrowHelperError(new ArgumentOutOfRangeException(nameof($1))

ThrowHelperError\(new ArgumentOutOfRangeException\(\"([a-zA-Z0-9]+)\",
ThrowHelperError(new ArgumentOutOfRangeException(nameof($1),

new ArgumentOutOfRangeException\(\"([a-zA-Z0-9]+)\",
new ArgumentOutOfRangeException(nameof($1),

ThrowHelperArgumentNull\(\"([a-zA-Z0-9]+)\"\)
ThrowHelperArgumentNull(nameof($1))
```